src/Toml/Edsl.hs

```diff
@@ -31,10 +39,18 @@ type TDSL = State TOML ()
 mkToml :: TDSL -> TOML
 mkToml env = execState env mempty
 
+-- | Creates an empty 'TDSL'.
+empty :: TDSL
+empty = put mempty
+
 -- | Adds key-value pair to the 'TDSL'.
 (=:) :: Key -> Value a -> TDSL
 (=:) k v = modify $ insertKeyVal k v
 
 -- | Adds table to the 'TDSL'.
 table :: Key -> TDSL -> TDSL
 table k = modify . insertTable k . mkToml
+
+-- | Adds array of tables to the 'TDSL'.
+array :: Key -> NonEmpty TDSL -> TDSL
```

Comment:
I feel like `array` is not the best name because it conflicts with ordinary arrays in TOML.

Reply:
`tableArray`?